src/libraries/System.Private.CoreLib/src/System/Runtime/InteropServices/NativeLibrary.cs

```diff
@@ -256,5 +256,26 @@ public static void SetDllImportResolver(Assembly assembly, DllImportResolver res
 
             return resolver(libraryName, assembly, hasDllImportSearchPathFlags ? (DllImportSearchPath?)dllImportSearchPathFlags : null);
         }
+
+        /// <summary>
+        /// Get a handle that can be used with <see cref="GetExport" /> or <see cref="TryGetExport" /> to resolve exports from the entry point module.
+        /// </summary>
+        /// <returns> The handle that can be used to resolve exports from the entry point module.</returns>
+        public static IntPtr GetEntryPointModuleHandle()
+        {
+            IntPtr result = IntPtr.Zero;
+#if TARGET_WINDOWS
+            result = Interop.Kernel32.GetModuleHandle(null);
+#else
+            result = Interop.Sys.GetDefaultSearchOrderPseudoHandle();
+#endif
+            // I don't know when a failure case can occur here, but checking for it and throwing an exception
+            // if we encounter it.
+            if (result == IntPtr.Zero)
+            {
+                Marshal.ThrowExceptionForHR(Marshal.GetHRForLastWin32Error());
```

Comment:
This still does not look right for non-Windows. This assumes that the Unix native PInvoke errors codes are the same as Win32 error codes which they certainly are not.

Reply:
Do we have a generalized mechanism today for converting a Unix `errno` value to an exception or a pattern I should follow for converting the error code to an exception on non-Windows platforms?

Reply:
> Do we have a generalized mechanism today for converting a Unix errno value to an exception

We do not. (The closest one is `GetExceptionForIoErrno`, but it is very IO scenario focused and it may not produce the right result here.)

I think we typically use `throw new Win32Exception()` in situation like this one that are expected to never throw in practice. Despite its name, Win32Exception on non-Windows treats the error code as the platform specific error code (not as Windows error code).